### PR TITLE
Newlines: rename {never,avoid}InResultType

### LINKED ISSUE #2179: Unable to format due to bug
This template is a guideline, not a strict requirement.
- **Version**: 2.6.4
- **Integration**: CLI
- **Configuration**:
```
version=2.6.4
newlines.neverInResultType = true
```

## Steps

Given code like this:
```scala
object a {
implicit protected val td: TildeArrow[RequestContext, Future[RouteResult]] {
    type Out = RouteTestResult
  } = TildeArrow.injectIntoRoute

  }
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt formats code like this:
```
org.scalafmt.cli.FailedToFormat: /private/tmp/a.scala
Caused by: org.scalafmt.Error$PreciseIncomplete: /private/tmp/a.scala:2: error: Unable to format file due to bug in scalafmt
implicit protected val td: TildeArrow[RequestContext, Future[RouteResult]] {
                                                                           ^
```

## Workaround

Disable `neverInResultType`.

## Notes

- [The only place](https://github.com/scalameta/scalafmt/blob/af2205e38716474c39a9a8933547f2cb8772e030/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala#L999-L1003) of config usage
- Also `neverInResultType` config is not documented yet.

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #2179.

### CODE REVIEW

File: scalafmt-tests/src/test/resources/unit/DefDef.stat [hunk 1]
```diff
@@ -156,3 +156,35 @@ val a = new a[b, c](/*c1*/d => e)(/*c2*/f => g, h => i)(/*c3*/j => k)
 val a = new a[b, c]( /*c1*/ d => e)(
     /*c2*/ f => g,
     h => i)( /*c3*/ j => k)
+<<< #2179 neverInResultType=true
+newlines.neverInResultType = true
```
Comment (poslegm) [inline scalafmt-tests/src/test/resources/unit/DefDef.stat:160]:
```suggestion
newlines.avoidInResultType = true
```

File: scalafmt-tests/src/test/resources/unit/DefDef.stat [hunk 2]
```diff
@@ -156,3 +156,35 @@ val a = new a[b, c](/*c1*/d => e)(/*c2*/f => g, h => i)(/*c3*/j => k)
...
+}
+>>>
+object a {
+  implicit protected val td
+      : TildeArrow[RequestContext, Future[RouteResult]] {
+        type Out = RouteTestResult
+      } = TildeArrow.injectIntoRoute
+}
+<<< #2179 neverInResultType=false
+newlines.neverInResultType = false
```
Comment (poslegm) [inline scalafmt-tests/src/test/resources/unit/DefDef.stat:175]:
```suggestion
newlines.avoidInResultType = false
```

**Review by poslegm — APPROVED** [review]
Thanks :)

File: scalafmt-tests/src/test/resources/unit/DefDef.stat (hunk 2 above)
Comment (kitbellew) [reply scalafmt-tests/src/test/resources/unit/DefDef.stat:175]: done

**Pull request merged**